src/tracy_impl.rs

```diff
@@ -1,11 +1,36 @@
 #[macro_export]
 macro_rules! scope {
-    ($name:expr) => {
+    // Note: literal patterns provided as an optimization since they can skip an allocation.
+    ($name:literal) => {
+        // Note: callstack_depth is 0 since this has significant overhead
+        let _tracy_span = $crate::tracy_client::span!($name, 0);
+    };
+    ($name:literal, $data:expr) => {
         // Note: callstack_depth is 0 since this has significant overhead
-        let _tracy_span = $crate::tracy_client::Span::new($name, "", file!(), line!(), 0);
+        let _tracy_span = $crate::tracy_client::span!($name, 0);
+        _tracy_span.emit_text($data);
+    };
+    ($name:expr) => {
+        let function_name = {
+            struct S;
+            let type_name = core::any::type_name::<S>();
+            &type_name[..type_name.len() - 3]
```

Comment:
maybe at some point we put this in a macro so we don't duplicate code (especially if we ever want to do this in a different backend), but it's fine the way it is now :)